Generalize recipe, clean up API, introduce eeg specific recipe

SimonDanisch opened the pull request:
- [x] add more tests
- [x] clean up interpolator API quite a bit 
- [x] generalize core topoplot recipe to make it work for different use cases 
- [x] add EEG recipe
- [x] add more docstrings
- [x] load default montage positions from file (implement `labels2positions`) 
- [x] actually test things with Percy and setup sane CI, clean up tests + add more
- [x] write docs + clean up docstrings
- [x] update README

Enables:

```julia
labels = string.(1:length(positions))
TopoPlots.eegtopoplot(data[:, 340, 1], labels; positions=positions, axis=(aspect=DataAspect(),))
```
<img width="466" alt="image" src="https://user-images.githubusercontent.com/1010467/175339668-56646201-1a1d-4484-bc44-2f0a41df98c1.png">

behinger commented:
great!

what I wondered, why is position required for recipe-topoplot, but label required for recipe-eegtopoplot?
Could one multiple dispatch on position::string->label vs position::Float -> positions? (feels too smart, so maybe not a good idea).
I just want to be sure, that label is the right default and not positions.

some pro / con regarding label
+ often 10/20 system is used => convenient if such a system
- labels are not strictly necessary, only positions
- individual positions

Other plotting libraries have a chanloc object / struct that contains both

Will give it a spin this week, this is just from inspecting the code!

one other thing: the topoplot_series, should that go into here, or stay in UnfoldMakie?

behinger commented:
great!

what I wondered, why is position required for recipe-topoplot, but label required for recipe-eegtopoplot?
Could one multiple dispatch on position::string->label vs position::Float -> positions? (feels too smart, so maybe not a good idea).
I just want to be sure, that label is the right default and not positions.

some pro / con regarding label
+ often 10/20 system is used => convenient if such a system
- labels are not strictly necessary, only positions
- individual positions

Other plotting libraries have a chanloc object / struct that contains both

Will give it a spin this week, this is just from inspecting the code!

one other thing: the topoplot_series, should that go into here, or stay in UnfoldMakie?

behinger commented:
ok. just a clarification: positions cannot always be looked up, e.g. custom locations, custom caps etc. - but o.c. for that the position argument exists. I'm ok with it.

I will keep plot_topoplot_series in UnfoldMakie for now

SimonDanisch commented:
>I will keep plot_topoplot_series in UnfoldMakie for now

Ah forgot to answer that... I think it's perfectly well placed in TopoPlots.jl

behinger commented:
some first observations:

`topoplot(position,data)`
whereas
`eegtopoplot(data,label)`

- [ ] I prefer the later - shall I change it?
- [ ] include("eeg.jl") is missing right now
- [ ] any reason the eegtopoplot recipe is not in recipe.jl?

SimonDanisch commented:
1) Just changed it :) 
2) Changed it already this morning ;) 
3) Wanted to rename recipes.jl going forward, but didn't have a good name yet... Just in case we end up having more recipes, it's nicer to have them in separate files

behinger commented:
can you push it? ^^

SimonDanisch commented:
Ended up debugging some weird Cairo artifact: https://github.com/JuliaPlots/Makie.jl/pull/2093
This will pop up in the refimages until we tag a new release with this.

behinger commented:
> It would probably be good to point users to `hidedecorations!` and `hidespines!` for "naked" head plots. (Speaking of... should we define a method for those with `FigureAxisPlot`? Right now, they're only defined on `Axis`.)

I noticed this back then, especially for many plots, hidedecorations&splines is slow: https://github.com/JuliaPlots/Makie.jl/issues/1335

right now i do:
 ```julia
axisOptions = (aspect = 1,xgridvisible=false,xminorgridvisible=false,xminorticksvisible=false,xticksvisible=false,xticklabelsvisible=false,xlabelvisible=false,ygridvisible=false,yminorgridvisible=false,yminorticksvisible=false,yticksvisible=false,yticklabelsvisible=false,ylabelvisible=false,
    leftspinevisible = false,rightspinevisible = false,topspinevisible = false,bottomspinevisible=false,)
```
which is much faster, but not very userfriendly (no merging of options as of now)

SimonDanisch commented on .github/workflows/CI.yml:
```diff
@@ -32,6 +32,10 @@ jobs:
       - uses: julia-actions/cache@v1
       - uses: julia-actions/julia-buildpkg@v1
       - uses: julia-actions/julia-runtest@v1
+      - name: Percy Upload
```
It's running: https://percy.io/e4642824/TopoPlots/builds/19246994/changed/1080394421?browser=edge&browser_ids=18%2C22%2C23%2C25&subcategories=unreviewed%2Cchanges_requested&viewLayout=overlay&viewMode=new&width=1000&widths=800%2C1000
But we still need to enable github integration, which needs a percy admin... @jrevels maybe?

palday replied:
I think that needs something done on the GH end, not the percy end, then. I'll see what I can do.
